From: Ben Hutchings <ben@decadent.org.uk>
Subject: alx: update for 3.8
Date: Mon, 25 Feb 2013 05:27:56 +0000
Bug-Debian: http://bugs.debian.org/699129
Forwarded: no

__devinit and __devexit are no longer defined or needed.

__netdev_printk() has been removed, so use "%pV" to pass a
va_format to netdev_printk().

--- a/drivers/net/ethernet/atheros/alx/alx_main.c
+++ b/drivers/net/ethernet/atheros/alx/alx_main.c
@@ -122,7 +122,7 @@ void alx_hw_printk(const char *level, co
 	vaf.va = &args;
 
 	if (hw && hw->adpt && hw->adpt->netdev)
-		__netdev_printk(level, hw->adpt->netdev, &vaf);
+		netdev_printk(level, hw->adpt->netdev, "%pV", &vaf);
 	else
 		printk("%salx_hw: %pV", level, &vaf);
 
@@ -1947,7 +1947,7 @@ static void alx_reset_interrupt_mode(str
 }
 
 
-static int __devinit alx_init_adapter_special(struct alx_adapter *adpt)
+static int alx_init_adapter_special(struct alx_adapter *adpt)
 {
 	switch (adpt->hw.mac_type) {
 	case alx_mac_l1f:
@@ -2002,7 +2002,7 @@ init_alf_adapter:
 /*
  * alx_init_adapter
  */
-static int __devinit alx_init_adapter(struct alx_adapter *adpt)
+static int alx_init_adapter(struct alx_adapter *adpt)
 {
 	struct alx_hw *hw   = &adpt->hw;
 	struct pci_dev	*pdev = adpt->pdev;
@@ -3450,8 +3450,8 @@ static const struct net_device_ops alx_n
 /*
  * alx_init - Device Initialization Routine
  */
-static int __devinit alx_init(struct pci_dev *pdev,
-		       const struct pci_device_id *ent)
+static int alx_init(struct pci_dev *pdev,
+		    const struct pci_device_id *ent)
 {
 	struct net_device *netdev;
 	struct alx_adapter *adpt = NULL;
@@ -3744,7 +3744,7 @@ err_alloc_device:
 /*
  * alx_remove - Device Removal Routine
  */
-static void __devexit alx_remove(struct pci_dev *pdev)
+static void alx_remove(struct pci_dev *pdev)
 {
 	struct alx_adapter *adpt = pci_get_drvdata(pdev);
 	struct alx_hw *hw = &adpt->hw;
@@ -3866,7 +3866,7 @@ static struct pci_driver alx_driver = {
 	.name        = alx_drv_name,
 	.id_table    = alx_pci_tbl,
 	.probe       = alx_init,
-	.remove      = __devexit_p(alx_remove),
+	.remove      = alx_remove,
 	.shutdown    = alx_shutdown,
 	.err_handler = &alx_err_handler,
 	.driver.pm   = ALX_PM_OPS,
